app/models/manageiq/providers/vmware/infra_manager/provision/configuration/disk.rb

```diff
@@ -21,13 +21,33 @@ def build_config_disk_spec(vmcs)
     end
   end
 
-  def get_scsi_controller_info
-    inventory_hash = source.with_provider_connection do |vim|
-      vim.virtualMachineByMor(source.ems_ref_obj)
+  def build_disk_relocate_spec(clone_options)
+      VimHash.new('VirtualMachineRelocateSpecDiskLocator') do |disk_locator|
```

Comment:
According to https://pubs.vmware.com/vsphere-6-5/index.jsp?topic=%2Fcom.vmware.vsan.apiref.doc%2Fvim.vm.RelocateSpec.html the `disk` attribute should be an array of `VirtualMachineRelocateSpecDiskLocator` so I think this should be switched:

```
(disks_info + get_new_disks).map do |disk|
  VimHash.new('VirtualMachineRelocateSpecDiskLocator') do |disk_locator|
  end
end
```

Reply:
Ugh, I was _so close_. Thanks.

Reply:
Yeah, VMware is unforgiving